overlord/snapstate/backend/setup.go

```diff
@@ -100,20 +109,43 @@ func (b Backend) SetupSnap(snapFilePath, instanceName string, sideInfo *snap.Sid
 		opts.MustNotCrossDevices = true
 	}
 
+	if setupOpts.IntegrityRootHash != "" {
+		integrityDataPath := integrity.DmVerityHashFileName(snapFilePath, setupOpts.IntegrityRootHash)
+		_, err := os.Stat(integrityDataPath)
+		if err != nil {
+			// setup a snap with integrity data was requested but there was an issue
+			// when accessing the file
+			return snapType, nil, fmt.Errorf("%w: %w", ErrDmVerityDataFileError, err)
+		}
+
+		opts.IntegrityRootHash = setupOpts.IntegrityRootHash
+
+		// The dm-verity type and the digest are the minimum information that are needed
+		// to mount a snap with dm-verity data as the dm-verity hash file path is composed
+		// from the snap file and the digest.
+		s.IntegrityData = &snap.IntegrityDataInfo{
+			IntegrityDataParams: integrity.IntegrityDataParams{
+				Type:   "dm-verity",
```

Comment:
shouldn't this come from setupOpts.Type as well? or is that is not really used atm?